### PR TITLE
Add ability to re-order slides with buttons

### PR DESCRIPTION
**Pull request by gnarf**
image

### CODE REVIEW

**Comment by gnarf** [comment]
after margin tweak: 
image

**Comment by gnarf** [comment]
and - only showing buttons on hover:
image

File: server/service/scenarios/slides/endpoints.js [hunk 1]
```diff
@@ -20,7 +20,10 @@ exports.addSlide = asyncMiddleware(async (req, res) => {
 
 exports.orderSlides = asyncMiddleware(async (req, res) => {
     const { id: scenario_id } = reqScenario(req);
-    const slide_ids = req.body.slides;
+    const slide_ids = req.body.slides.map(id => {
+        if (typeof id === 'object') return id.id;
```
Comment (evmiguel) [inline server/service/scenarios/slides/endpoints.js]: When would id be an object? Out of curiosity, non-blocking

File: client/components/Editor/Slides.jsx [hunk 2]
```diff
@@ -157,21 +186,50 @@ class Slides extends React.Component {
...
+                                                        index,
+                                                        index - 1
+                                                    )
+                                                }
+                                            />
+                                            <Button
+                                                icon="arrow alternate circle down outline"
+                                                aria-label="Move down"
+                                                disabled={
+                                                    index === slides.length - 1
```
Comment (evmiguel) [inline client/components/Editor/Slides.jsx]: Clever

File: client/components/Editor/Slides.jsx [hunk 3]
```diff
@@ -157,21 +186,50 @@ class Slides extends React.Component {
...
+                                                    />
+                                                );
                                             }
                                         )}
-                                        <p>
+                                        <div className="Slides-button-bar">
+                                            <Button
+                                                icon="arrow alternate circle up outline"
+                                                aria-label="Move up"
+                                                disabled={index === 0}
```
Comment (evmiguel) [inline client/components/Editor/Slides.jsx]: Clever!

File: server/service/scenarios/slides/endpoints.js (hunk 1 above)
Comment (gnarf) [reply server/service/scenarios/slides/endpoints.js]: lets you submit `{"slides":[{}, {},{}]}` with ids in the objects, or just the ids of the slides

**Pull request merged**